tests/phpunit/tests/dependencies/styles.php

```diff
@@ -517,4 +517,98 @@ public function test_block_styles_for_viewing_with_split_styles() {
 			$GLOBALS['wp_styles']->registered['wp-block-library']->src
 		);
 	}
+
+	/**
+	 * @ticket 58394
+	 *
+	 * @covers ::wp_maybe_inline_styles
+	 */
+	public function test_wp_maybe_inline_styles() {
+		wp_register_style( 'test-handle', '/' . WPINC . '/css/classic-themes.css' );
+		wp_style_add_data( 'test-handle', 'path', ABSPATH . WPINC . '/css/classic-themes.css' );
+
+		wp_enqueue_style( 'test-handle' );
+
+		wp_maybe_inline_styles();
+
+		$this->assertFalse( $GLOBALS['wp_styles']->registered['test-handle']->src, 'Source of style should be reset to false' );
+		$this->assertNotEmpty( $GLOBALS['wp_styles']->registered['test-handle']->extra['after'], 'Source of style should set to after property' );
+	}
+
+	/**
+	 * @ticket 58394
+	 *
+	 * @covers ::wp_maybe_inline_styles
+	 */
+	public function test_wp_maybe_inline_styles_multiple_runs() {
+		$filter = new MockAction();
+		add_filter( 'pre_wp_filesize', array( $filter, 'filter' ) );
+		wp_register_style( 'test-handle', '/' . WPINC . '/css/classic-themes.css' );
+		wp_style_add_data( 'test-handle', 'path', ABSPATH . WPINC . '/css/classic-themes.css' );
+
+		wp_enqueue_style( 'test-handle' );
+
+		wp_maybe_inline_styles();
+
+		$this->assertFalse( $GLOBALS['wp_styles']->registered['test-handle']->src, 'Source of style should be reset to false' );
+		$this->assertNotEmpty( $GLOBALS['wp_styles']->registered['test-handle']->extra['after'], 'Source of style should set to after property' );
+
+		wp_maybe_inline_styles();
+
+		$this->assertSame( 1, $filter->get_call_count(), 'wp_filesize should only be called once' );
+	}
+
+	/**
+	 * @ticket 58394
+	 *
+	 * @covers ::wp_maybe_inline_styles
+	 */
+	public function test_wp_maybe_inline_styles_invalid() {
```

Comment:
What behavior is this method attempting to test? There seems to be several different things being tested. Could those be separated?

Reply:
Agreed. I'd also advise to use a more precise method name, the "invalid" isn't clear. Invalid _what_?

Reply:
This tests for invalid url / path combo. I want to test for a "unhappy path", as in when the there is something wrong. This test ensure if the file does not exist, then a error is not triggered. Core does not do enough "unhappy path" testing IMO.

Reply:
If helpful, I can change the name of the test. Any suggestions?

Reply:
@spacedmonkey Maybe `test_test_wp_maybe_inline_styles_missing_file()`?